### PR TITLE
Add Modernbert sequence packing

### PR DESCRIPTION
**Pull request by galopyz**
This PR adds sequence unpadding and repadding with flash attention 2 varlen for the attention and flash attention rope. Inside of the model, attention_mask is created and sequences are unpadded before passing to decoder layers. After all the decoder layers and final norm, the unpadded sequences are padded again before passing to lm_head. We repad for the loss function calculation. In the future, we can get rid of repadding and do loss calculation ourselves as _unpad_modernbert_input also returns unpadded labels.

Results for sdpa (green) vs. FA2 (red) on wandb: https://wandb.ai/local-research-group/smollm2-135m-training-avelina/workspace?nw=nwusergalopyz. Grey one uses FA2 with batch size of 8 instead of 4. Because there are no padding tokens, I could fit 8bs inside of RTX 3090. (24GB RAM)

This PR also adds ModernBert sequence packing. With sequence packing, we can pack sequences together into a packed sequence of size `[int(batch_size / micro_batch_size), int(micro_batch_size * max_seq_len)]`. This ensures we are feeding fixed number of tokens every microbatch, unlike just unpadding approach where microbatches can have varying number of tokens depending on the length of the sequences. 

There are also some minor changes:

- Also used variables for micro batch size and batch size in yaml.

- To use sequence packing, `numba` is added.

- Disabled aim logger.

- Added full fine tuning option to `custom_model_training.py`.

### CODE REVIEW

**Review by vishalbakshi — APPROVED** [review]
@galopyz and I have both run multiple successful training runs (locally and on Modal) using this branch (`modernbert_sequence_packing`).
